lxd/images.go

```diff
@@ -1989,9 +1989,44 @@ func pruneExpiredImagesTask(d *Daemon) (task.Func, task.Schedule) {
 
 func pruneLeftoverImages(d *Daemon) {
 	opRun := func(op *operations.Operation) error {
+		// Check if dealing with shared image storage.
+		var storageImages string
+		err := d.State().Node.Transaction(func(tx *db.NodeTx) error {
+			nodeConfig, err := node.ConfigLoad(tx)
+			if err != nil {
+				return err
+			}
+
+			storageImages = nodeConfig.StorageImagesVolume()
+
+			return nil
+		})
+		if err != nil {
+			return err
+		}
+
+		if storageImages != "" {
+			// Parse the source.
+			fields := strings.Split(storageImages, "/")
```

Comment:
@stgraber please can you move this logic and error handling into a function in `daemon_storage.go` called something like `parseDaemonStorageVolumeName()` or similar, as this will be the 5th time that this logic and error handling has been duplicated (from a quick look).